backend/src/middleware/notifications/notificationsMiddleware.spec.js

```diff
@@ -541,5 +571,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given I file a report on a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-23T18:42:57Z" title="Thursday, January 23rd 2020, 7:42:57 pm +01:00">Jan 23, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -541,5 +573,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given me filing a report of a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
+        return expect.objectContaining({
+          data: {
+            notifications: [
+              {
+                createdAt: expect.any(String),
+                read: false,
+                reason: 'filed_report_on_resource',
+                from: {
+                  __typename: 'Report',
+                  id: expect.any(String),
+                  filed: [
+                    {
+                      reasonCategory: 'discrimination_etc',
+                      reasonDescription: 'I am free to be gay !!!',
+                      reportedResource,
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        reportFiler = notifiedUser
+        reportedUserOrAuthorData = {
+          id: 'reportedUser',
+          name: 'Mrs Badman',
+          slug: 'mrs-badman',
+          email: 'reported-user@example.org',
+          password: '1234',
+        }
+      })
+
+      describe('user', () => {
+        it('sends me a notification for filing a report on a user', async () => {
+          await neode.create('User', reportedUserOrAuthorData)
+          resourceId = 'reportedUser'
+          await fileReportAction()
+
+          const expected = setExpectedNotificationOfReportedResource({
+            __typename: 'User',
+            id: 'reportedUser',
+            name: 'Mrs Badman',
+          })
+          const { query } = createTestClient(server)
+          await expect(
+            query({
+              query: notificationQuery,
+              variables: {
+                read: false,
+              },
+            }),
+          ).resolves.toEqual(expected)
+        })
+      })
+
+      describe('post and comment', () => {
+        beforeEach(async () => {
+          title = 'My post'
+          postContent = 'My post content.'
+          postAuthor = await neode.create('User', reportedUserOrAuthorData)
+        })
+
+        describe('post', () => {
+          it('sends me a notification for filing a report on a post', async () => {
+            await createPostAction()
+            resourceId = 'p47'
+            await fileReportAction()
+
+            const expected = setExpectedNotificationOfReportedResource({
+              __typename: 'Post',
+              id: 'p47',
+              content: postContent,
+            })
+            const { query } = createTestClient(server)
+            await expect(
+              query({
+                query: notificationQuery,
+                variables: {
+                  read: false,
+                },
+              }),
+            ).resolves.toEqual(expected)
+          })
+        })
+
+        describe('comment', () => {
+          beforeEach(async () => {
+            commentContent = 'Commenters comment.'
```


```suggestion
            commentContent = 'Commenter's comment.'
```

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-01-24T09:27:23Z" title="Friday, January 24th 2020, 10:27:23 am +01:00">Jan 24, 2020</time>_

---


Thank you!